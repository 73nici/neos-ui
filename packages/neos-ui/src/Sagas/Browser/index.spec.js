import {expectSaga} from 'redux-saga-test-plan';
import {watchContentURIChange, reflectChangeInAddressBar} from './index.js';

describe('watchContentURIChange()', () => {
    it('should be a generator function.', () => {
        expect(typeof watchContentURIChange).toBe('function');
        expect(typeof watchContentURIChange().next).toBe('function');
    });

    it('should takeEvery SET_CONTEXT_PATH actionType and execute the reflectChangeInAddressBar generator.', async function() {
        const result = await expectSaga(watchContentURIChange).run();

        expect(result.toJSON()).toMatchSnapshot();
    });
});

describe('reflectChangeInAddressBar()', () => {
    let replaceState;

    beforeEach(() => {
        replaceState = jest.spyOn(history, 'replaceState').mockImplementation(jest.fn());
    });

    afterEach(() => {
        replaceState.mockRestore();
    });

    it('should be a generator function.', () => {
        expect(typeof reflectChangeInAddressBar).toBe('function');
        expect(typeof reflectChangeInAddressBar().next).toBe('function');
    });

    it('should call the history.replaceState method with the actions payload "contextPath" property.', async function() {
        const result = await expectSaga(reflectChangeInAddressBar, {type: 'DATA', payload: {contextPath: 'fooContextPath'}}).run();

        expect(result.toJSON()).toMatchSnapshot();
    });
});
