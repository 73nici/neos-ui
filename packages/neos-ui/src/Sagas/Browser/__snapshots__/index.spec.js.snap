// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reflectChangeInAddressBar() should call the history.replaceState method with the actions payload "contextPath" property. 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          Object {},
          "",
          "?node=fooContextPath",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/mockConstructor",
      },
    },
  ],
}
`;

exports[`watchContentURIChange() should takeEvery SET_CONTEXT_PATH actionType and execute the reflectChangeInAddressBar generator. 1`] = `
Object {
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "TAKE": Object {
        "pattern": "@neos/neos-ui/UI/ContentCanvas/SET_CONTEXT_PATH",
      },
    },
  ],
}
`;
